sortinghat/core/decorators.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2014-2019 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Santiago Dueñas <sduenas@bitergia.com>
+#     Miguel Ángel Fernández <mafesan@bitergia.com>
+#
+
+from django.conf import settings
+from graphql_jwt.decorators import user_passes_test
+
+auth_required = user_passes_test(
```

Comment:
I think we should rename this function to something like `check_auth` or similar. I'm saying this because the name does something different that the function really does. If `settings.AUTHENTICATION_REQUIRED` is deactivated, the function evaluates to True, so you are saying that authentication is not required but when this is active, you are checking if the user is authenticated.

I also think we should document this somehow or we are going to forget why we're doing this this way. I think it will be enough if you define this as:

```
def auth_required:
    return user_passes_test(
        lambda u: u.is_authenticated or not settings.AUTHENTICATION_REQUIRED
    )
```
Inside the function you can explain that this works as a decorator and how to use it.

Reply:
You are right, I have renamed `auth_required` to `check_auth`. Moreover, I have added a comment explaining briefly what this decorator does.

Reply:
About defining the decorator as proposed, I was having difficulties defining it as a method because it received the calling method as positional argument.

In this case, the method would have to be defined as:

```python
def check_auth(func):
    return user_passes_test(
        lambda u: u.is_authenticated or not settings.AUTHENTICATION_REQUIRED
    )(func)
```
@sduenas and me have agreed on keeping this decorator as it was for the moment, but adding the comments I mentioned above.